pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -713,7 +713,7 @@ func (nc *Controller) monitorNodeHealth() error {
 		}
 
 		// We do not treat a master node as a part of the cluster for network disruption checking.
-		if !system.IsMasterNode(node.Name) {
+		if !system.IsMasterNode(node) {
```

Comment:
This was wrong before, but I'm not sure using the label makes it correct.

I'm not sure this check is correct for self hosted masters, although admittedly it's been a long time since I looked at this code.

I think we may ultimately need a specific label that must be set on nodes to tell things like this code and service load balancer to ignore this node, not key off the role.